src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -177,6 +177,12 @@ public bool RequireLicenseAcceptance
             set;
         }
 
+        public bool EmitRequireLicenseAcceptance
```

Comment:
Thanks for the update! Apologies for the late response.

We should add a condition that throws when a license is required but EmitRequireLicenseAcceptance is set to false. Otherwise. You can end up with packages requiring a license but no RequireLicenseAcceptance tag added.

See test below for PackageBuilder:

```csharp
[Theory]
           [InlineData(true, true, true)]
        // [InlineData(true, true, false)] // invalid: license is required
           [InlineData(true, false, false)]
        // [InlineData(true, false, true)] // invalid: license not required
        // [InlineData(false, true, true)] // invalid: license is required
           [InlineData(false, true, false)] // This case should throw
           [InlineData(false, false, false)]
        // [InlineData(false, false, true)] // invalid: license not required
        public void EmitLicenseAcceptance_Suceeds(bool emitRequireLicense, bool requireLicense, bool result)
        {
            var license = NuGetLicenseExpression.Parse("MIT");
            var list = new string[] { };
            var builder = new PackageBuilder()
            {
                Id = "test",
                Version = new NuGetVersion("0.0.1"),
                Authors = {
                    "TestAuthors"
                },
                Description = "Test package for EmitRequireLicenseAcceptance",
                EmitRequireLicenseAcceptance = emitRequireLicense,
                RequireLicenseAcceptance = requireLicense,
                LicenseMetadata = requireLicense ?
                    new LicenseMetadata(LicenseType.Expression, "MIT", license, list, LicenseMetadata.EmptyVersion):
                    null,
                DependencyGroups =
                {
                    new PackageDependencyGroup(
                        targetFramework: NuGetFramework.Parse("netstandard1.4"),
                        packages: new[]
                        {
                            new PackageDependency("another.dep", VersionRange.Parse("0.0.1"))
                        })
                }
            };

            using (var testDir = TestDirectory.Create())
            {
                var packagePath = Path.Combine(testDir, "test.0.0.1.nupkg");

                using (FileStream nupkgStream = File.Create(packagePath))
                {
                    builder.Save(nupkgStream);
                }

                var reader = new PackageArchiveReader(packagePath);

                Assert.Equal(result, reader.NuspecReader.GetRequireLicenseAcceptance());

                using (var nureader = new StreamReader(reader.GetNuspec()))
                {
                    var nuspecContent = nureader.ReadToEnd();

                    if (result)
                    {
                        Assert.Contains("<requireLicenseAcceptance>true</requireLicenseAcceptance>", nuspecContent);
                    }
                }
            }
        }
```

Reply:
@dominoFire Would this be something for PackageBuilderTest.cs or NuGetPackCommandTest.cs (where it looks like you copied from)?

Reply:
@jnm2 I think it is for PackageBuilderTests.cs, since it is testing PackageBuilder . Thanks for reaching out!

Reply:
Added.